source/databricks/calculation_engine/tests/calculation_logic/features/energy_calculations/positive_grid_loss/when_period_is_one_hour/states/scenario_state.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2020 Energinet DataHub A/S
+#
+# Licensed under the Apache License, Version 2.0 (the "License2");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from datetime import datetime
+
+from pyspark.sql import DataFrame, SparkSession
+
+import calculation_logic.utils as cl
+import calculation_logic.utils.factories as clf
+from package.codelists import (
+    QuantityQuality,
+    InputMeteringPointType,
+    InputSettlementMethod,
+    MeteringPointResolution,
+)
+
+
+def get_expected(*args) -> DataFrame:
+    """
+    This function can be used to custom build the expected results (dataframe).
+    It is also used a reference to locate the test scenario.
+    """
+    return cl.create_energy_result_dataframe(*args)
+
+
+def create_grid_loss_metering_points(spark: SparkSession) -> DataFrame:
+    factory = clf.InputGridLossTestFactory(spark)
+    row1 = factory.create_row(metering_point_id="571313180400100657")
+    row2 = factory.create_row(metering_point_id="571313180480500149")
+    return factory.create_dataframe([row1, row2])
+
+
+def create_time_series_points(spark: SparkSession) -> DataFrame:
+    factory = clf.InputTimeSeriesPointTestFactory(spark)
+    row1 = factory.create_row(
+        metering_point_id="571313180400010437",
+        quantity=10.0,
+        quality=QuantityQuality.MEASURED.value,
+        observation_time=datetime(2023, 2, 1, 12, 0, 0),
+    )
+    row2 = factory.create_row(
+        metering_point_id="571313180400010673",
+        quantity=5.0,
+        quality=QuantityQuality.MEASURED.value,
+        observation_time=datetime(2023, 2, 1, 12, 0, 0),
+    )
+    row3 = factory.create_row(
+        metering_point_id="571313180400140417",
+        quantity=10.0,
+        quality=QuantityQuality.MEASURED.value,
+        observation_time=datetime(2023, 2, 1, 12, 0, 0),
+    )
+    return factory.create_dataframe([row1, row2, row3])
+
+
+def create_metering_point_periods(spark: SparkSession) -> DataFrame:
+    factory = clf.InputMeteringPointPeriodsTestFactory(spark)
+    row1 = factory.create_row(
```

Comment:
It would be easier to read with meaningful names. E.g.
```suggestion
    consumption_period = factory.create_row(
```
or "consumption = ..."

Reply:
Actually, I think this is way better - took me a while to figure out:
```suggestion
    positive_grid_loss = factory.create_row(
```

Reply:
But this leads to another thought:
```suggestion
    positive_grid = factory.create_positive_grid_loss_row(...)
```
where the following input parameters are removed:
- metering_point_type, settlement_method